scripts/setup_offline_updates.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+
+usage() {
+    echo
+    echo "Usage: $0 <repo-type> <tar-file>"
+    echo
+    echo "Creates or updates an XCP-ng offline <repo-type> repository from the <tar-file> archive."
+    echo
+    echo "Arguments:"
+    echo "  repo-type   The repository type, based on its name on online mirrors. Example : 'updates'."
+    echo "  tar-file    The path to the archive containing the repository files."
+}
+
+if [ -z "$2" ]; then
+    echo "Error: argument(s) missing."
+    usage
+    exit 1
+fi
+
+set -eu
+
+REPO_TYPE="$1"
+TAR_FILE="$2"
+REPOS_DIR="/var/local/xcp-ng-repos"
+REPONAME="xcp-ng-offline-$REPO_TYPE"
+
+if [ ! -e "$TAR_FILE" ]; then
+    echo "Error: file $TAR_FILE not found."
+    exit 2
+fi
+
+if ! tar -tf "$TAR_FILE" | grep -q "^${REPO_TYPE}/$"; then
+    echo "Error: directory $REPO_TYPE/ not found in archive $TAR_FILE."
+    exit 3
+fi
+
+mkdir -p "$REPOS_DIR"
+
+if [ -d "$REPOS_DIR/$REPO_TYPE" ]; then
+    echo "- Deleting old repo at $REPOS_DIR/$REPO_TYPE"
+    rm "$REPOS_DIR/$REPO_TYPE" -r
+fi
+
+echo "- Extracting $TAR_FILE to $REPOS_DIR/$REPO_TYPE"
+tar -xf "$TAR_FILE" -C "$REPOS_DIR"
+
+echo "- Writing /etc/yum.repos.d/$REPONAME.repo"
+cat <<EOF > "/etc/yum.repos.d/$REPONAME.repo"
+[xcp-ng-offline-$REPO_TYPE]
+name=XCP-ng Offline ${REPO_TYPE^} Repository
+baseurl=file://$REPOS_DIR/$REPO_TYPE/x86_64/
+enabled=1
+gpgcheck=1
```

Comment:
any reason not to use `repo-gpgcheck`?

Reply:
Clearly looks like I'm missing context, then

Reply:
It's an update repo which only contains the latest of each package, and I haven't signed the new repodata, but I could if we think it's useful.

Reply:
Actually I do produce signatures for the repo metadata :sweat_smile:. I added `repo_gpgcheck`.